google/cloud/storage/client.h

```diff
@@ -3084,6 +3085,7 @@ class Client {
         SpanOptions(std::forward<Options>(options)...));
     internal::PolicyDocumentV4Request request(std::move(document));
     request.set_multiple_options(std::forward<Options>(options)...);
+    request.SetEndpointAuthority(EndpointAuthority());
```

Comment:
I kept it in [default constructor](https://github.com/googleapis/google-cloud-cpp/pull/14179/files#diff-3349e5e6da87ae44561dfdd606132f2cc49554b9beac03b89dcb2b90c4e34cc0R108) and provided a setter, just like `scheme` . I wanted to do similar in `SignUrlRequest` as well but I couldn't do that there because I was putting the new variable in `SignUrlRequestCommon`.

Reply:
But I'm not sure which one is the right / better approach. :D

Reply:
Fair.  In general, immutable objects (or at least fields) are easier to reason about.  I think we would prefer to create the request with (as much of) its state set correctly.